[BUG] container builds should fail verbosely when CLI not in PATH

#### What happened:
Executing `make microshift` without `podman` or`docker` CLI installed fails opaquely, without a descriptive error.  Instead, the given error is just `/bin/sh: line 1: build: command not found`.

#### What you expected to happen:
`make microshift` and other container builds should explicitly report the binary dependency is missing.

#### How to reproduce it (as minimally and precisely as possible):

1. w/o `podman` || `docker` in $PATH: `$ make microshift`

#### Environment:

- OS:
  - Fedora release 34 (Thirty Four)
  - Darwin redhat 18.7.0 Darwin Kernel Version 18.7.0 
#### Relevant Logs 

Failed run:
```
❯ make microshift
make _build_containerized ARCH=amd64
make[1]: Entering directory '/home/jcope/Workspace/github.com/redhat-et/microshift'
echo BIN_TIMESTAMP==2021-08-17T20:31:04Z
BIN_TIMESTAMP==2021-08-17T20:31:04Z
build -t quay.io/microshift/microshift:4.7.0-0.microshift-2021-08-05-215756-3-g04ef964-linux-amd64 \
        -f "/home/jcope/Workspace/github.com/redhat-et/microshift"/images/build/Dockerfile \
        --build-arg SOURCE_GIT_TAG=4.7.0-0.microshift-2021-08-05-215756-3-g04ef964 \
        --build-arg BIN_TIMESTAMP=2021-08-17T20:31:04Z \
        --build-arg ARCH=amd64 \
        --build-arg MAKE_TARGET="cross-build-linux-amd64" \
        --platform="linux/amd64" \
        .
/bin/sh: line 1: build: command not found
make[1]: *** [Makefile:86: _build_containerized] Error 127
make[1]: Leaving directory '/home/jcope/Workspace/github.com/redhat-et/microshift'
make: *** [Makefile:97: build-containerized-cross-build-linux-amd64] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -20,8 +20,6 @@ IMAGE_REPO :=quay.io/microshift/microshift
 OUTPUT_DIR :=_output
 CROSS_BUILD_BINDIR :=$(OUTPUT_DIR)/bin
 
-CTR_CMD :=$(or $(shell which podman 2>/dev/null), $(shell which docker 2>/dev/null))
-
 GO_EXT_LD_FLAGS :=-extldflags '-static'
 GO_LD_FLAGS :=-ldflags "-X k8s.io/component-base/version.gitMajor=1 \
                    -X k8s.io/component-base/version.gitMinor=20 \
@@ -81,7 +79,9 @@ cross-build: cross-build-linux-amd64 cross-build-linux-arm64
 ###############################
 # containerized build targets #
 ###############################
+_build_containerized: CTR_CMD :=$(or $(shell which podman 2>/dev/null), $(shell which docker 2>/dev/null))
 _build_containerized:
+	@if [ -z '$(CTR_CMD)' ] ; then echo '!! ERROR: containerized builds require podman||docker CLI, none found $$PATH' >&2 && exit 1; fi
 	echo BIN_TIMESTAMP==$(BIN_TIMESTAMP)
 	$(CTR_CMD) build -t $(IMAGE_REPO):$(SOURCE_GIT_TAG)-linux-$(ARCH) \
 		-f "$(SRC_ROOT)"/images/build/Dockerfile \
@@ -105,6 +105,10 @@ build-containerized-cross-build:
 	+$(MAKE) build-containerized-cross-build-linux-amd64
 	+$(MAKE) build-containerized-cross-build-linux-arm64
 .PHONY: build-containerized-cross-build
+
+###############################
+# dev targets                 #
+###############################
 
 vendor:
 	./hack/vendoring.sh

